pyfar/signals/deterministic.py

```diff
@@ -242,6 +298,155 @@ def exponential_sweep(n_samples, frequency_range, n_fade_out=90, amplitude=1,
     return signal
 
 
+def exponential_sweep_freq(
+        n_samples, start_margin=None, stop_margin=None, frequency_range=None,
+        butterworth_order=8, double=True, sampling_rate=44100):
+
+    signal, group_delay = _sweep_synthesis_freq(
+        n_samples, "exponential", start_margin, stop_margin,
+        frequency_range, butterworth_order, double, sampling_rate)
+
+    return signal, group_delay
+
+
+def magnitude_weighted_sweep(
+        n_samples, magnitude, start_margin=None, stop_margin=None,
+        double=True, sampling_rate=44100):
+
+    signal, group_delay = _sweep_synthesis_freq(
+        n_samples, magnitude, start_margin, stop_margin,
+        None, None, double, sampling_rate)
+
+    return signal, group_delay
+
+
+def perfect_sweep(
```

Comment:
I'd rather keep only perfect, because I think it is an established term. How about mentioning that it is a special case of a linear sweep at the beginning of the docstring instead?